rs/index/src/multi_spann/builder.rs

```diff
@@ -115,15 +150,19 @@ impl MultiSpannBuilder {
         &self.base_directory
     }
 
-    /// This function will remove the builder for the given user id.
+    /// This function will remove and return the SPANN builder for the given user id.
     /// If the builder is not found, it will return None.
     pub fn take_builder_for_user(&self, user_id: u128) -> Option<SpannBuilder> {
         self.inner_builders
             .remove(&user_id)
-            .map(|builder| builder.1.into_inner().unwrap())
+            .map(|builder| builder.1.into_inner())
             .or(None)
     }
 
+    pub fn bloom_filter(&self) -> &RwLock<Option<BlockedBloomFilter>> {
+        &self.bloom_filter
+    }
```

Comment:
I'm not a fan of returning ref to a lock. I think in this case, we only take the bloomfilter to write it to disk, so let's just do something like `take_builder_for_user`, you just take the ownership of object and return, replace the current value with None.